### PR TITLE
fix: default vault client to using internal kubernetes service dns na…

### PR DESCRIPTION
**Pull request by rawlingsj**
…me when running incluster

### CODE REVIEW

File: pkg/kube/vault/vault.go [hunk 1]
```diff
@@ -430,17 +432,22 @@ func GetVaults(client kubernetes.Interface, vaultOperatorClient versioned.Interf
...
-		if err != nil {
-			log.Logger().Warnf("error finding vault service url setting to empty string, err: %s", err)
-			vaultURL = ""
-		}
-		if vaultURL == "" {
-			vaultURL, err = services.FindIngressURL(client, ns, vaultName)
+		// default to using internal kubernetes service dns name
+		vaultURL := fmt.Sprintf(defaultInternalVaultURL, vaultName)
+		// is there a better way to do this?
+		if os.Getenv("KUBERNETES_SERVICE_HOST") == "" {
```
Comment (rawlingsj) [inline pkg/kube/vault/vault.go]: wonder if this is ewnough to check that we're running incluster seeing that we dont have the common options factory in these functions?

Comment (pmuir) [reply pkg/kube/vault/vault.go]: It should be easy to refactor the `IsInCluster()` method off the factory and into a util method as it depends on precisely nothing

**Comment by rawlingsj** [comment]
gonna add an interface and mock around `cluster.IsInCluster()` so it's easier to test as we will get different behaviour when running tests in or outside of a pipeline

/hold

**Pull request merged**